packages/ember-runtime/lib/system/core_object.js

```diff
@@ -109,6 +109,7 @@ function makeCtor(base) {
                 property === 'didAddListener' ||
                 property === 'didRemoveListener' ||
                 property === 'isDescriptor' ||
+                property === '_onLookup' ||
```

Comment:
I thought we were just going to exclude type 'function' in addition to undefined

Reply:
We did that, but as you can see in this case the `unknownProperty` _always_ returns non-null non-undefined.

Reply:
doesn't it return a function?

Reply:
Not on the instance which is what this test is doing. The test replicates how ember-engines passes down services from the host app into the engines own container. Roughly akin to:


    instance = hostAppOwner.lookup(name);
    engineOwner.register(name, instance, { instantiate: false })

`container.lookup` uses `factoryFor` first (so it can later call .create()) and that does the `_onLookup` probe on the result (an instance in this case).